### ISSUE TITLE
`tsx >4.11.2` is broken when running `vite`

### ISSUE THREAD

**Issue by o-alexandrov** [privatenumber/tsx#590]
### Acknowledgements

- [X] I read the documentation and searched existing issues to avoid duplicates
- [X] I am reporting a confirmed bug (not seek debugging help)
- [X] I understand this is a collaborative project and relies on my contributions
- [X] I read and understood the [Contribution guide](https://github.com/privatenumber/tsx/blob/develop/CONTRIBUTING.md)

### Minimal reproduction URL

https://github.com/o-alexandrov/bug-tsx-vite

### Problem & expected behavior (under 200 words)

To reproduce:

1. clone example & install deps
2. run `npm run start`

see error in cli:

```sh
tsx ./node_modules/vite/bin/vite.js

import { defineConfig } from "vite";

export default defineConfig({});

tsx ./node_modules/vite/bin/vite.js

failed to load config from ./vite.config.ts
error when starting dev server:
TypeError: Cannot assign to read only property '.ts' of object '[object Object]'
    at loadConfigFromBundledFile (./node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:66923:36)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at loadConfigFromFile (./node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:66754:24)
    at resolveConfig (./node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:66364:24)
    at _createServer (./node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:62977:18)
    at CAC.<anonymous> (./node_modules/vite/dist/node/cli.js:735:20)
```

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.15.5

If you rely on this project, please consider [making a donation](https://github.com/sponsors/privatenumber) :heart: :pray:

**Comment by privatenumber** [comment]
Appreciate you always filing great bug reports @o-alexandrov 🙌
